sdk/core/core/inc/az_str.h

```diff
@@ -30,6 +30,9 @@
 // `void some_function(az_const_span const span);`
 #define AZ_STR(STRING_LITERAL) (az_const_span) AZ_CONST_STR(STRING_LITERAL)
 
+#define AZ_ZERO_STR AZ_STR("\0")
+#define AZ_HEADER_SEPARATOR_STR AZ_STR(": ")
```

Comment:
AZ_HEADER_SEPARATOR shouldn't be in az_str.h - it is too specific to HTTP requests.
In any case, there is no need to use #defines for these constants. For example, do create "extern az_const_span const AZ_ZERO_STR;", and create a file called az_str.c, and put AZ_ZERO_STR there. See how I did that AZ_HTTP_VERB_*

Reply:
AZ_STR_* is probably a better name than AZ_*_STR, i.e. AZ_STR_ZERO instead of AZ_ZERO_STR.

Reply:
BTW, @sergey-shandar, should we also define AZ_STR_CRLF the same way?